podman_hpc/hook_tool.py

```diff
@@ -62,10 +76,67 @@ def ldconfig():
         log(f"ldconfig failed: {ret}")
 
 
-def resolve_src(src, modulesd=os.path.abspath("")):
-    if not os.path.isabs(src):
-        src = os.path.join(modulesd, src)
-    return os.path.abspath(os.path.expandvars(src))
+def resolve_src_and_dest(rule, root_path, modulesd=os.path.abspath("")):
+    # extract source and destination patterns from the rule
+    rs, rd = (rule + ":").split(":")[:2]
+
+    # expand vars
+    rs = os.path.expanduser(os.path.expandvars(rs))
```

Comment:
Do we know if this actually works inside a hook execution?

Reply:
Also we redo this later.  So should we remove this one?

Reply:
@lastephey and verified that this patch works with our current modules on Muller.  

As for `expanduser` and `expandvars` specifically, I think you are right and they wouldn't work inside the hook unless those environment variables were passed in specifically.  But anyway, they won't make anything _not_ work that would have worked before.

Reply:
Oh, also, it's necessary to do the expands first, because otherwise the `os.path.isabs()` doesn't work correctly.  We could remove the second call, but it would be necessary if for some reason `modulesd` contained a variable or tilde.

Reply:
Agree, we did test `mpi` `gpu` `cvmfs` and `cuda-mpi` functionality and they looks ok. Still working on getting a test for `nccl`.

Reply:
Ah, good point!  Should we wait for a nccl test before merging?

Reply:
No, I say let's go with what we have or we might not make the maintenance.